libpirate/ge_eth.c

```diff
@@ -256,11 +268,11 @@ int pirate_ge_eth_close(ge_eth_ctx *ctx) {
     return rv;
 }
 
-ssize_t pirate_ge_eth_read(const pirate_ge_eth_param_t *param, ge_eth_ctx *ctx, void *buf, size_t count) {
+ssize_t pirate_ge_eth_read(const pirate_ge_eth_param_t *param, ge_eth_ctx *ctx,
+                            void *buf, size_t count) {
     ssize_t rd_size;
-    ge_header_t hdr;
+    ge_header_t hdr = { 0 };
```

Comment:
This doesn't work on compilers that are using an older C standard. Fallback to memset that always works.

Reply:
This will also work if you don't want to memset:

 = { 0, 0, 0 };